core-bundle/src/Resources/contao/library/Contao/DcaExtractor.php

```diff
@@ -418,7 +418,7 @@ protected function createExtract()
 
 					if (isset($config['foreignKey']))
 					{
-						$table = substr($config['foreignKey'], 0, strrpos($config['foreignKey'], '.'));
+						$table = explode('.', $config['foreignKey'])[0];
```

Comment:
This is a behavior change. If a foreign key is `foo.bar.baz`, the table name was previously `foo.bar` and now it is only `foo`. What is the motivation for this change?

Reply:
See the comment here: https://github.com/contao/contao/pull/3321#discussion_r689708367

Also you cannot create a DCA called `tl_foo.bar` for example, as this would lead to the following exception:

```
Execute database migrations
---------------------------

 * CREATE TABLE tl_foo.bar (id INT UNSIGNED AUTO_INCREMENT NOT NULL, tstamp INT UNSIGNED DEFAULT 0 NOT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB ROW_FORMAT = DYNAMIC......FAILED


 [OK] Executed 0 SQL queries.



 [ERROR] An exception occurred while executing 'CREATE TABLE tl_foo.bar (id INT UNSIGNED AUTO_INCREMENT
         NOT NULL, tstamp INT UNSIGNED DEFAULT 0 NOT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4
         COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB ROW_FORMAT = DYNAMIC':

SQLSTATE[42000]: Syntax
         error or access violation: 1049 Unknown database 'tl_foo'
```